pyccel/ast/cwrapper.py

```diff
@@ -0,0 +1,176 @@
+        self._pykwarg    = python_func_kwargs
+        self._parse_args = parse_args
+        self._arg_names  = arg_names
+        self._flags      = ''
+        i = 0
+        while i < len(c_func_args) and not isinstance(c_func_args[i], ValuedVariable):
+            self._flags += pytype_parse_registry[(c_func_args[i].dtype, c_func_args[i].precision)]
+            i+=1
+        if i < len(c_func_args):
+            self._flags += '|'
+        while i < len(c_func_args):
+            self._flags += pytype_parse_registry[(c_func_args[i].dtype, c_func_args[i].precision)]
+            i+=1
+
+        # Restriction as of python 3.8
+        if any([isinstance(a, Variable) and a.is_kwonly for a in c_func_args]):
+            errors.report('Kwarg only arguments without default values will not raise an error if they are not passed',
+                          symbol=c_func_args, severity='warning')
+
+    @property
+    def pyarg(self):
+        return self._pyarg
+
+    @property
+    def pykwarg(self):
+        return self._pykwarg
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @property
+    def args(self):
+        return self._parse_args
+
+    @property
+    def arg_names(self):
+        return self._arg_names
+
+class PyBuildValueNode(Basic):
+    def __init__(self, result_args = []):
+        self._flags = '' 
+        self._result_args = result_args
+        for i in result_args:
+            self._flags += pytype_parse_registry[(i.dtype, i.precision)]
+
+    @property
+    def flags(self):
+        return self._flags
+    
+    @property
+    def args(self):
+        return self._result_args
+
+class CastFunction(FunctionDef):
+    """Represents a cast function definition."""
+
+    def __init__(self,
+        name,
+        cast_type,
+        arguments,
+        ret,
+        results):
+        self._name = name
+        self._arguments = arguments
+        self._ret = ret
+        self._results = results
+        self._cast_type = cast_type
+        body_ = ''
+        #TODO I dont know if the build of body shoudl be done here or in the cwrapper
+        #TODO this is just a tmp way of printing this shoudlbe improved later
+        if self._cast_type == 'pyint_to_bool':
+            body_ += '{} = {} != 0;\n'.format(self.results[0].name, self._arguments[0].name)
+
+        elif self.cast_type == 'bool_to_pyobj': 
+            body_ += '{} = {} != 0 ? Py_True : Py_False;\n'.format(self.results[0].name, self._arguments[0].name)
+        self._body = body_
+
+
+    def __hash__(self):
+        return  hash(self._cast_type)
+
+    def __eq__(self, other):
+        return (self.cast_type == other.cast_type)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def arguments(self):
+        return self._arguments
+
+    @property
+    def results(self):
+        return self._results
+
+    @property
+    def ret(self):
+        return self._ret
```

Comment:
Why does ret need to be a property? Could it not be appended to the body at the end of the initialisation?

Reply:
Yes, its true like I said in the comment didn't know how to implement the body of the cast_function class so I put it aside until I find a better way